mixnet/client/src/sender.rs

```diff
@@ -133,21 +133,60 @@ where
             return None;
         }
 
+        // corner case: 0 average delay
+        if self.avg_delay == 0 {
+            self.remaining_delays -= 1;
+            return Some(0);
+        }
+
         if self.remaining_delays == 1 {
             self.remaining_delays -= 1;
             return Some(self.remaining_time);
         }
 
+        // if we have no remaining time, then just return 0
+        if self.remaining_time == 0 {
+            self.remaining_delays -= 1;
+            return Some(0);
+        }
+
         // Calculate bounds to avoid extreme values
         let lower_bound = (self.avg_delay as f64 * 0.5) as u64;
         let upper_bound = (self.avg_delay as f64 * 1.5)
-            .min((self.remaining_time - self.remaining_delays + 1) as f64)
+            .min((self.remaining_time as f64 - self.remaining_delays as f64) + 1.0)
             as u64;
 
+        // corner case: when lower bound is greater than upper bound
+        if lower_bound >= upper_bound {
+            self.remaining_delays -= 1;
+            self.remaining_time = 0;
+            return Some(self.remaining_time);
+        }
```

Comment:
I'm thinking about the way to prevent this corner case, since this is not very clear to me :) but couldn't find the way yet.